src/openMVG/geometry/io/ply.cpp

```diff
@@ -0,0 +1,700 @@
+#include "ply.hpp"
+#include "ply_helper.hpp"
+
+#include <algorithm>
+#include <cctype>
+#include <fstream>
+#include <functional>
+#include <iterator>
+#include <locale>
+#include <string>
+
+namespace openMVG
+{
+namespace geometry
+{
+  namespace io
+  {
+    /**
+    * @brief Export a set of points to a PLY file 
+    * @param pts A list of points 
+    * @param nor A list of normal (one for each points) 
+    * @param col A list of color (one for each points) 
+    * @param path Path of the file to save 
+    * @param binary Indicate if the file should be written to binary format 
+    * @retval true if load is successful 
+    * @retval false if load fails 
+    */
+    bool PLYWrite( const std::vector<Vec3> &pts,
+                   const std::vector<Vec3> *nor,
+                   const std::vector<Vec3uc> *col,
+                   const std::string &path,
+                   const bool binary )
+    {
+      std::ofstream file( path );
```

Comment:
I don't know exactly, because it's mixed : text and binary. I think the filesystem manage it correcly depending on the underlying function (operator<<() or write()). We may need to check what the standard say about that.